### ISSUE TITLE
Use content negotiation to determine return type

### ISSUE THREAD

**Issue by blootsvoets** [RADAR-CNS/RADAR-RestApi#40]
Instead of changing the path, Avro should be returned if a client explicitly requests for it using an Accept header, otherwise JSON should be returned.

**Comment by yatharthranjan** [comment]
Should this be implemented using runtime content negotiation of JAX-RS ? 
Something like - 
```java
@GET
@Path("/" + GET_SUBJECT + "/{" + SUBJECT_ID + "}")
public Response get(@Context Request r) { 
    List<Variant> vs = Variant.mediaTypes(MediaType.APPLICATION_JSON_TYPE,
                MediaType.APPLICATION_OCTET_STREAM).build();
    Variant v = r.selectVariant(vs);
    MediaType reqMediaType = v.getMediaType();
    if (v == null) {
        return Response.notAcceptable(vs).build();
    } else if (reqMediaType.isCompatible(MediaType.APPLICATION_JSON_TYPE)) {
        return jsonSubjectResponse();
    } else if (reqMediaType.isCompatible(MediaType.APPLICATION_OCTET_STREAM)) {
        return avroSubjectResponse();
    }
    return Response.serverError().status(Response.Status.NOT_ACCEPTABLE).build();
}
```

**Comment by dennyverbeeck** [comment]
I think you can use the `@Produces` annotation and keep the path the same, e.g.:

```
@GET
@Path("/" + GET_SUBJECT + "/{" + SUBJECT_ID + "}")
@Produces(MediaType.APPLICATION_JSON)
public Response getJson(@Context Request r) {
   // produce some json here
}

@GET
@Path("/" + GET_SUBJECT + "/{" + SUBJECT_ID + "}")
@Produces("avro/binary")
public Response getJson(@Context Request r) {
   // produce some avro here
}
```

FYI: the [Avro spec](https://avro.apache.org/docs/1.8.2/spec.html) lists `avro/binary` as the HTTP content type to use for avro messages
> The HTTP Content-Type of requests and responses should be specified as "avro/binary"

**Comment by yatharthranjan** [comment]
@dennyverbeeck thanks for the info. This looks good. I think the method name is a typo and I assume it will be different for both the methods.
Also I was following the current implementation of REST-API in which the `MediaType.APPLICATION_OCTET_STREAM` is used as the return type of Avro based methods instead of avro/binary

**Comment by yatharthranjan** [comment]
In any case, both `application/octet-stream` and `avro/binary` return a binary file in avro format. So i guess use of any of the two is okay. But I agree that for better exposure of what is returned, `avro/binary` should be used.

**Comment by yatharthranjan** [comment]
closed by #50
